portal/static/js/main.js

```diff
@@ -2909,34 +2920,36 @@ $(document).ready(function() {
 });
 
 var tnthDates = {
-    /** validateBirthDate  check whether the date is a sensible date.
+    /** validateDateInputFields  check whether the date is a sensible date in month, day and year fields.
+     ** params: month, day and year values and error field ID
      ** NOTE this can replace the custom validation check; hook this up to the onchange/blur event of birthday field
      ** work better in conjunction with HTML5 native validation check on the field e.g. required, pattern match  ***/
-    "validateBirthDate": function(m, d, y) {
+    "validateDateInputFields": function(m, d, y, errorFieldId) {
         if (hasValue(m) && hasValue(d) && hasValue(y)) {
 
             var m = parseInt(m);
             var d = parseInt(d);
             var y = parseInt(y);
+            var errorField = $("#" + errorFieldId);
 
             if (!(isNaN(m)) && !(isNaN(d)) && !(isNaN(y))) {
                 var today = new Date();
                 // Check to see if this is a real date
                 var date = new Date(y,m-1,d);
                 if (!(date.getFullYear() == y && (date.getMonth() + 1) == m && date.getDate() == d)) {
-                    $("#errorbirthday").html("Invalid date. Please try again.").show();
+                    errorField.html("Invalid date. Please try again.").show();
                     return false;
                 }
                 else if (date.setHours(0,0,0,0) >= today.setHours(0,0,0,0)) {
-                    $("#errorbirthday").html("Birthday must not be in the future. Please try again.").show();
+                    errorField.html("Birthday must not be in the future. Please try again.").show();
```

Comment:
Should these error messages still reference "Birthday" if it's a more generic date validation function?